Source/pfile.cpp

```diff
@@ -102,7 +102,9 @@ BOOL pfile_create_player_description(char *dst, DWORD len)
 	_uiheroinfo uihero;
 
 	myplr = 0;
+#if !defined(__BIG_ENDIAN__)
 	pfile_read_player_from_save();

```

Comment:
As best I can tell the issue is with `LoadMpqTable`, as you can see here the file look up failes because the hash tables do not match when compared by `GetFirstHashEntry`:

https://github.com/diasurgical/devilutionX/blob/8d907499296ba4612595272f0157886c2dd1dd62/3rdParty/StormLib/src/SBaseCommon.cpp#L703

x86:
```
szFileName: hero
dwHashIndexMask: 0x000007FF 2047
dwIndex: 0x000003D3 979
pHash->dwName1: 0x33AF486B 867125355
dwName1: 0x33AF486B 867125355
pHash->dwName2: 0xE3D1C651 -472791471
dwName2: 0xE3D1C651 -472791471
pHash->dwBlockIndex: 0x00000003 3
ha->dwFileTableSize: 0x00000800 2048
```

PowerPC:
```
szFileName: hero
dwHashIndexMask: 0x000007FF 2047
dwIndex: 0x000003D3 979
pHash->dwName1: 0xC7F14F34 -940486860
dwName1: 0x33AF486B 867125355
pHash->dwName2: 0x3278DBDF 846781407
dwName2: 0xE3D1C651 -472791471
pHash->dwBlockIndex: 0x9C181A00 -1676142080
ha->dwFileTableSize: 0x00000800 2048
```
If you can help fix this then I'm happy to fix the byte order of the actual save game data for the load and save functions.

Reply:
This is how far I came as well. But when I started swapping around stuff there, it wasn't able to read any mpq file any more.

Reply:
Same here. 
Found only  that this can be safely swapped:
`void SHA1ProcessMessageBlock(SHA1Context *context)
{
	DWORD i, temp;
	DWORD W[80];
	DWORD A, B, C, D, E;

	DWORD *buf = (DWORD *)context->buffer;

	for (i = 0; i < 16; i++) {
		W[i] = SDL_SwapLE32(buf[i]);`

Casting char to int causes byte swap on Big Endian.

Save game cannot be opened at all in
```
handle = open(szFileName, oflag | O_LARGEFILE);
INVALID_HANDLE_VALUE line 159 in file 3rdParty/StormLib/src/FileStream.cpp
```

Reply:
I'm happy to report that I've made progress and am able to go in-game with save file from x86! :D
With pfile_read_player_from_save();  enabled :)

Still to do is fix creating of save file.

Reply:
That's awesome! :D

Reply:
@arczi84 Oh, that's fantastic news :D

Reply:
Good news everyone!
https://usercontent.irccloud-cdn.com/file/6Sbo9JKN/VID_20191231_201327.mp4

Reply:
@arczi84 that's amazing perfect timing also as I just did an upstream release, meaning it's a good time to merge new code :) Looking very much forward to you pushing the code.

Reply:
I'll upload when  everything works.

Reply:
Ok I'm happy to help out though :)

Reply:
Ok, so saving and loading works if inventory has not changed.

Reply:
I tried merging the new code, but on my PowerPC creating a new game simpley segfaults, is this expected? Do i need a PC save game for things to work properly?